posthog/models/property.py

```diff
@@ -37,47 +53,114 @@
     "is_date_before",
 ]
 
+OperatorInterval = Literal["day", "week", "month", "year"]
 GroupTypeName = str
 PropertyIdentifier = Tuple[PropertyName, PropertyType, Optional[GroupTypeIndex]]
 
 NEGATED_OPERATORS = ["is_not", "not_icontains", "not_regex", "is_not_set"]
 CLICKHOUSE_ONLY_PROPERTY_TYPES = ["static-cohort", "precalculated-cohort"]
 
+VALIDATE_PROP_TYPES = {
+    "event": ["key", "value"],
+    "person": ["key", "value"],
+    "cohort": ["key", "value"],
+    "element": ["key", "value"],
+    "static-cohort": ["key", "value"],
+    "precalculated-cohort": ["key", "value"],
+    "group": ["key", "value", "group_type_index"],
+    "recording": ["key", "value"],
+    "performed_event": ["event", "event_type", "time_value", "time_interval"],
+    "performed_event_multiple": ["event", "event_type", "time_value", "time_interval", "operator", "operator_value"],
+}
+
 
 class Property:
-    key: str
+    key: Optional[str]
     operator: Optional[OperatorType]
-    value: ValueT
+    value: Optional[ValueT]
     type: PropertyType
     group_type_index: Optional[GroupTypeIndex]
 
+    event_type: Optional[str]
+    event: Optional[Union[str, int]]
```

Comment:
I think we can keep the `key` non optional by getting rid of `event` ? It's kinda been so for existing properties, where the key is action_id / event name / etc.

Also, would love a comment on what each of these parameters mean!

Also wondering if all different kind of sequences/lifecycle need their own property type. It does feel like an operator, rather than a completely new type.